src/libs/actions/PersonalDetails.ts

```diff
@@ -163,75 +158,81 @@ function updateDisplayName(firstName, lastName) {
                         [currentUserAccountID]: {
                             firstName,
                             lastName,
-                            displayName: getDisplayName(currentUserEmail, {
+                            displayName: getDisplayName(currentUserEmail ?? '', {
                                 firstName,
                                 lastName,
                             }),
                         },
                     },
                 },
             ],
-        },
-    );
+        });
+    }
+
     Navigation.goBack(ROUTES.SETTINGS_PROFILE);
 }
 
-/**
- * @param {String} legalFirstName
- * @param {String} legalLastName
- */
-function updateLegalName(legalFirstName, legalLastName) {
-    API.write(
-        'UpdateLegalName',
-        {legalFirstName, legalLastName},
-        {
-            optimisticData: [
-                {
-                    onyxMethod: Onyx.METHOD.MERGE,
-                    key: ONYXKEYS.PRIVATE_PERSONAL_DETAILS,
-                    value: {
-                        legalFirstName,
-                        legalLastName,
-                    },
+function updateLegalName(legalFirstName: string, legalLastName: string) {
+    type UpdateLegalNameParams = {
```

Comment:
i'm not sure about the footprint, but can we try extracting this declaration out of a function?

Reply:
We are going to move all these types to a proper place in the future ;)